CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CIReinforcementsManager.uc

```diff
@@ -0,0 +1,174 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Manager class to control the delay of any reinforcements issued
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_CIReinforcementsManager extends XComGameState_BaseObject;
+
+var array<DelayedReinforcementSpawner> DelayedReinforcementSpawners;
+
+var protectedwrite bool bNeedsUpdate;
+var const int Threshold;
+
+var localized string m_strReinforcementsBodyWarning;
+var localized string m_strReinforcementsBodyImminent;
+
+static function CreateReinforcementsManager()
+{
+	local XComGameState NewGameState;
+	local XComGameState_Player PlayerState;
+	local Object ThisObj;
+
+	if (GetReinforcementsManager(true) == none)
+	{
+		NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Creating reinforcements manager singleton");
+
+		NewGameState.CreateNewStateObject(class'XComGameState_CIReinforcementsManager');
+
+		`TACTICALRULES.SubmitGameState(NewGameState);
+	}
+
+	PlayerState = class'XComGameState_Player'.static.GetPlayerState(eTeam_XCom);
+	ThisObj = GetReinforcementsManager();
+	`XEVENTMGR.RegisterForEvent(ThisObj, 'PlayerTurnBegun', OnPlayerTurnBegun, ELD_OnStateSubmitted, , PlayerState);
```

Comment:
Why filter by `PlayerState`?

Reply:
otherwise it gets called 5 extra times for no reason